### PR TITLE
[MO][Kaldi] Add Statistics Extracting and Pooling components

### PR DESCRIPTION
**Pull request by pavel-esir**
Added support of Kaldi statistics extracting and pooling components

ticket #22512

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR: N/A -- original model has fixed `batch_size=1` and is not reshape-able 
* [x]  Transformation preserves original framework node names

Validation:
* [x]  Unit tests
* [x]  Framework operation tests: -- manual preliminary check gives the same results with the original Kaldi model
* [x]  Transformation tests: N/A
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: N/A
* [x]  Model Optimizer IR Reader check

Documentation:
* [x]  Supported frameworks operations list
* [x]  Supported **public** models list: N/A
* [x]  New operations specification: N/A
* [x]  Guide on how to convert the **public** model: N/A
* [x]  User guide update: N/A

### CODE REVIEW

File: model-optimizer/extensions/ops/statistics.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from mo.graph.graph import Graph
+from mo.ops.op import Op
+
+class KaldiStatisticsExtraction(Op):
+    op = 'statisticsextractioncomponent'
+
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/statistics.py:22]: enabled = False

File: model-optimizer/extensions/ops/statistics.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            'in_ports_count': 1,
+            'out_ports_count': 1,
+            'infer': None,
+            'input_dim': None,
+        }, attrs)
+
+
+class KaldiStatisticsPooling(Op):
+    op = 'statisticspoolingcomponent'
+
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/statistics.py:36]: enabled = False

File: model-optimizer/extensions/ops/statistics.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+from mo.graph.graph import Graph
+from mo.ops.op import Op
+
+class KaldiStatisticsExtraction(Op):
+    op = 'statisticsextractioncomponent'
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/ops/statistics.py:21]: Please, add some comments about this operation and what MO does with it. 

The same comment for the operation below.

File: model-optimizer/extensions/front/kaldi/replace_stats_extract_pool.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+from mo.graph.graph import Node, Graph
+from mo.ops.assign import Assign
+from mo.ops.concat import Concat
+from mo.ops.const import Const
+from mo.ops.read_value import ReadValue
+from mo.ops.result import Result
+import numpy as np
+
+class StatsExtractPoolReplacer(FrontReplacementSubgraph):
+    enabled = True
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/kaldi/replace_stats_extract_pool.py:29]: Provide explanation of the transformation.

File: model-optimizer/extensions/front/kaldi/replace_stats_extract_pool.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+
+    def run_after(self):
+        from extensions.front.kaldi.memory_offset_adjustment import MemoryOffsetAdjustment
+
```
Comment (lazarevevgeny) [inline model-optimizer/extensions/front/kaldi/replace_stats_extract_pool.py:33]: remove redundant empty line

**Review by lazarevevgeny — CHANGES_REQUESTED** [review]

**Comment by pavel-esir** [comment]
Closing, @sadolini is finishing these changes on her fork

**Pull request closed without merging**